File: 1413_MinimumValueToGetPositiveStepByStepSum/minimumValueToGetPositiveSteBySteSum.js
var minStartValue = function(nums) {
    let minVal = 0
    let total = 0

    for(let i = 0; i < nums.length; i++) {
    	total += nums[i]
    	minVal = Math.min(minVal, total)
    }

    return -minVal + 1
};

console.log(minStartValue([-3,2,-3,4,2]))
console.log(minStartValue([1,2]))
console.log(minStartValue([1,-2,-3]))

File: 1413_MinimumValueToGetPositiveStepByStepSum/minimumValueToGetPositiveSteBySteSum.py
def minStartValue(self, nums: List[int]) -> int:
    total = minVal = 0

    for i in range(len(nums)):
        total += nums[i]
        minVal = min(minVal, total)

    return -minVal + 1

print(minStartValue([-3,2,-3,4,2]))
print(minStartValue([1,2]))
print(minStartValue([1,-2,-3]))